### PR TITLE
Dev refactoringwizard closes dev-151, dev-168, dev-123

### PR DESCRIPTION
**Pull request by babkenmes**

### CODE REVIEW

File: components/Wizard/Steps/ExternalPaymentStep.tsx [hunk 1]
```diff
@@ -3,80 +3,81 @@ import { useSwapDataState, useSwapDataUpdate } from '../../../context/swap';
...
-            goToStep("Email")
+            goToStep(SwapWithdrawalStep.Email)
         const swap = await getSwap(swapId.toString())
-        const { payment } = swap?.data || {}
-        //TODO handle no payment url
-        const { payment_url } = payment.external_flow_context || {}
+        const payment_url = swap?.data?.additonal_data?.payment_url
         window.open(payment_url, '_blank', 'width=420,height=720')
-    }, [])
+    }, [network])
```
Comment (babgev) [inline components/Wizard/Steps/ExternalPaymentStep.tsx:53]: Why would the network get changed here?

File: components/Wizard/Steps/MainStep.tsx [hunk 2]
```diff
@@ -327,27 +148,13 @@ export default function MainStep() {
...
-        availableNetworks.forEach(x => {
-            if (x != initialNetwork)
-                x.isEnabled = false;
-        })
-    }
-
-    const initialValues: SwapFormValues = generateSwapInitialValues(formValues?.swapType ?? "onramp", settings, query, account, chainId)
+    const initialValues: SwapFormValues = generateSwapInitialValues(formValues?.swapType ?? SwapType.OnRamp, settings, query, account, chainId)
 
     const exchangeRef: any = useRef();
```
Comment (babgev) [inline components/Wizard/Steps/MainStep.tsx:159]: What are these refs for?

File: components/Wizard/Steps/MainStep.tsx [hunk 3]
```diff
@@ -391,7 +198,7 @@ export default function MainStep() {
                                 </div>
                             </div>
                             {
-                                values.swapType === "onramp" &&
+                                values.swapType === SwapType.OnRamp && (()=>{console.log(values.swapType);return true})() &&
```
Comment (babgev) [inline components/Wizard/Steps/MainStep.tsx:201]: console

File: components/Wizard/Steps/OverviewStep.tsx [hunk 4]
```diff
@@ -19,40 +23,39 @@ const OverviewStep: FC = () => {
...
+
+                const exchange = exchanges.find(e => e.currencies.some(ec => ec.id === swap.data.exchange_currency_id))
+
+                if (swapStatus == SwapStatus.Completed)
+                    goToStep(SwapWithdrawalStep.Success)
+                else if (swapStatus == SwapStatus.Failed || swapStatus == SwapStatus.Cancelled || swapStatus === SwapStatus.Expired)
+                    goToStep(SwapWithdrawalStep.Failed)
+                else {
+                    if (swap?.data?.type === SwapType.OffRamp)
+                        goToStep(SwapWithdrawalStep.OffRampWithdrawal) ///TODO only for coinbase, implement other flows
```
Comment (babgev) [inline components/Wizard/Steps/OverviewStep.tsx:45]: idk reminding for the todo :)

File: components/Wizard/Steps/WithdrawExhangeStep.tsx [hunk 5]
```diff
@@ -18,48 +18,46 @@ import WarningMessage from '../../WarningMessage';
...
-            else if (payment?.status == "completed")
-                goToStep("Processing")
-            // else if (swapStatus == SwapStatus.Pending)
-            //     await goToStep("Processing")
-        }
+        if (currentStep !== SwapWithdrawalStep.Withdrawal)
+            return true;
+
+        const authData = TokenService.getAuthData();
+        if (!authData)
```
Comment (babgev) [inline components/Wizard/Steps/WithdrawExhangeStep.tsx:37]: Isn't this same code repeated in a lot of places?

File: components/exchangesComponent.tsx [hunk 6]
```diff
@@ -196,23 +196,26 @@ function UserExchanges() {
                                                 <Combobox.Option
                                                     key={item.id}
                                                     value={item}
-                                                    disabled={!item.is_enabled || !item.is_enabled}
-                                                    className={`bg-darkblue-500 ${!item.is_enabled ? 'opacity-35 cursor-not-allowed' : ''}  select-none rounded-lg p-3`}
+                                                    disabled={false}
```
Comment (babgev) [inline components/exchangesComponent.tsx:199]: Is this intentional? e.g. for testing?

File: components/exchangesComponent.tsx [hunk 7]
```diff
@@ -226,11 +229,11 @@ function UserExchanges() {
                                                             </div>
                                                             <div className="text-xs">
                                                                 {
-                                                                    item.authorization_flow && item.authorization_flow !== "none" && item.is_enabled && exchangeLoading?.id !== item.id &&
+                                                                    item.authorization_flow && item.authorization_flow !== "none" && exchangeLoading?.id !== item.id &&
```
Comment (babgev) [inline components/exchangesComponent.tsx:232]: enabled check

File: components/SwapOptionsToggle.tsx [hunk 8]
```diff
@@ -1,32 +1,35 @@
...
     } = useFormikContext<SwapFormValues>();
+    const { account, chainId } = useWeb3React<Web3Provider>();
 
     const settings = useSettingsState()
     const query = useQueryState()
     const name = 'swapType'
 
-    const handleFieldChange = (value: SwapType) => {
-        const initialValues = generateSwapInitialValues(value, settings, query)
+    const handleFieldChange = useCallback((value: SwapType) => {
```
[the remark this answers is not in the document]
Comment (babkenmes) [reply components/SwapOptionsToggle.tsx:28]: As all the fields control their handlers we should change them also, Main steps code can get bigger from that. I think we should address this if we reuse the field components and have some wrapper that passes handlers.

**Pull request merged**